File: Client/tecmo-wager/src/app/core/models/bettable-game.model.ts
export interface BettableGameOdds {
  spread: number;
  favoredPlayerId: number | null;
  overUnder: number | null;
  moneyLinePlayer1: number | null;
  moneyLinePlayer2: number | null;
  /** Present when the house posted an LLM/analyst blurb. */
  summary?: string;
}

/** Pending dollars per side; matches API marketDepth (camelCase). */
export interface BettableGameMarketDepth {
  maxMarketImbalance: number;
  spreadPlayer1: number;
  spreadPlayer2: number;
  over: number;
  under: number;
  moneyLinePlayer1: number;
  moneyLinePlayer2: number;
}

export interface WagerActionItem {
  playerName: string;
  side: string;
  stakeAmount: number;
}

export interface BettableGame {
  gameResultId: number;
  tournamentId: number;
  player1Id: number;
  player2Id: number;
  player1Name: string;
  player2Name: string;
  player1ProfilePic: number;
  player2ProfilePic: number;
  /** Server GameStatus name (Waiting, InProgress, Completed). */
  gameStatus?: string;
  /** True when API allows new wagers on this game. */
  isOpenForBetting?: boolean;
  odds: BettableGameOdds;
  gameStartedAt: string | null;
  /** Set for completed games from the games board API */
  player1Score?: number | null;
  player2Score?: number | null;
  marketDepth?: BettableGameMarketDepth;
  action?: WagerActionItem[];
}

File: Service/Models/BettableGameModel.cs
namespace TecmoTourney.Models
{
    /// <summary>Game available to bet: active tournament, status Waiting, GameStartedAt null, has odds with GameResultId.</summary>
    public class BettableGameModel
    {
        public int GameResultId { get; set; }
        public int TournamentId { get; set; }
        public int Player1Id { get; set; }
        public int Player2Id { get; set; }
        public string Player1Name { get; set; } = string.Empty;
        public string Player2Name { get; set; } = string.Empty;
        public int Player1ProfilePic { get; set; }
        public int Player2ProfilePic { get; set; }
        /// <summary>Final scores for completed games; otherwise null.</summary>
        public int? Player1Score { get; set; }
        public int? Player2Score { get; set; }
        public BettableGameOddsModel Odds { get; set; } = new();
        /// <summary>When set, betting is closed.</summary>
        public DateTime? GameStartedAt { get; set; }
        /// <summary>Who wagered what on this game (when ShowActionOnGames is true).</summary>
        public List<WagerActionItemModel>? Action { get; set; }
        /// <summary>Aggregated pending dollars per side; always populated for liquidity / imbalance UI.</summary>
        public BettableGameMarketDepthModel MarketDepth { get; set; } = new();
    }
}
